app/models/agents/naive_bayes_agent.rb

```diff
@@ -0,0 +1,509 @@
+require 'yaml'
+
+# NBayes from Oasic Technologies
+# https://github.com/oasic/nbayes
+#
+# Robust implementation of NaiveBayes:
+# - using log probabilities to avoid floating point issues
+# - Laplacian smoothing for unseen tokens
+# - allows binarized or standard NB
+# - allows Prior distribution on category to be assumed uniform (optional)
+# - generic to work with all types of tokens, not just text
+
+
+module NBayes
```

Comment:
Is this a copy of the code from the gem? It might make more sense to do this Agent as it's own gem using [huginn_agent](https://github.com/cantino/huginn_agent) and require the nbayes from it's gemspec.